src/game/xstate/api/fetchRedditPics.ts

```diff
@@ -41,5 +42,12 @@ export default async function fetchRedditPics(request: MediaRequest) {
     });
   });
 
+  for (const link of res.links) {
+    if (!(link.mediaType.toLowerCase() === "video" && link.sourceLink && link.directLink.includes("redgifs"))) continue;
+    const i = res.links.indexOf(link);
+    const redGifID = link.directLink.split("/")[3].split("-")[0];
+    res.links[i] = await searchRedGifs(redGifID.toLowerCase());
+  }
+
```

Comment:
Lets update this snippet so that it resolves all redgifs asynchronously  instead of await for each iteration.  It should make it several times faster!

```suggestion
  const redGifLinks = await resolveRedgifsLinks(
    res.links.filter((link) => link.directLink.includes("redgifs"))
  );
  const links = res.links.filter(
    (link) => !link.directLink.includes("redgifs")
  );

  return [...links, ...redGifLinks];

```

```ts
// Asynchronously resolve signed urls for redgif links.
async function resolveRedgifsLinks(redGifLinks: MediaLink[]) {
  const redgifsSignedLinksPromise = await Promise.allSettled(
    redGifLinks.map((link) => {
      const redGifID = link.directLink.split("/")[3].split("-")[0];
      return searchRedGifs(redGifID.toLowerCase());
    })
  );

  const redgifsSignedLinks = redgifsSignedLinksPromise
    .filter(
      (linkPromise): linkPromise is { status: "fulfilled"; value: MediaLink } =>
        linkPromise.status === "fulfilled"
    )
    .map((link) => link.value);

  return redgifsSignedLinks;
}
```

Reply:
This is awesome! Thank you! Just added your changes :)